npm/ipsm/ipsm.go

```diff
@@ -291,6 +305,11 @@ func (ipsMgr *IpsetManager) DeleteFromSet(setName, ip string) error {
 		return err
 	}
 
+	// Now cleanup the cache
+	if _, exists := ipsMgr.setMap[setName].elements[ip]; exists {
```

Comment:
I think this condition check has already been done in line 285, we can go ahead and cleanup the cache.